### PR TITLE
fix: deprecated `Node.js 16 actions`

### LINKED ISSUE #130: `Node.js 16 actions are deprecated.`

### PR DESCRIPTION
**Pull request by hamirmahal**
fixes #130.

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -10,26 +10,25 @@ name: Java CI with Maven
...
+          cache: maven
+      - name: Build with Maven
+        run: mvn -B package --file pom.xml
 
-    # Optional: Uploads the full dependency graph to GitHub to improve the quality of Dependabot alerts this repository can receive
-    - name: Update dependency graph
-      uses: advanced-security/maven-dependency-submission-action@571e99aab1055c2e71a1e2309b9691de18d6b7d6
+      # Optional: Uploads the full dependency graph to GitHub to improve the quality of Dependabot alerts this repository can receive
+      - name: Update dependency graph
+        uses: advanced-security/maven-dependency-submission-action@v4.0.0
```
Comment (hamirmahal) [inline .github/workflows/maven.yml:34]: We have to use `@v4.0.0` here, specifically, and not `@v4`, due to some issues with this GitHub Action at the time of this writing.

**Comment by hamirmahal** [comment]
# Without this change

image

# With this change

image

**Review by breed — APPROVED** [review]
thanks hamir!

**Comment by hamirmahal** [comment]
You're welcome, Ben!

**Pull request merged**